CenterForOpenScience/ember-osf-web
# [EMB-331] Send page views to Keen

## Deployment notes!
Need to update the config, adding `OSF.keenConfig`
```ts
OSF: {
    // ...
    keenConfig: {
        public: { // required for "popular pages" chart
            projectId: '...',
            writeKey: '...',
        },
        private: { // maybe optional?
            projectId: '...',
            writeKey: '...',
        },
    },
}
```

## Purpose
Send page view events to Keen


## Summary of Changes
- Add keen metrics adapter (adapted from [ember-osf](https://github.com/CenterForOpenScience/ember-osf/blob/develop/addon/metrics-adapters/keen.js))
- Make the `analytics` service a little less google-analytics-specific
- Make sure page view events are actually sent on analytics and forks pages
- Config changes:
  - Add `OSF.keenConfig`
  - Remove `OSF.keenProjectId`
  - There are several cookie names scattered around, group them together under `OSF.cookies`:
  ```
           cookies: {
               status: statusCookie,
               keenUserId: 'keenUserId',
               keenSessionId: 'keenSessionId',
               analyticsDismissAdblock: 'adBlockDismiss',
           },
  ```


## Side Effects / Testing Notes
One way to test this, end-to-end:
1. Go to a project's analytics page, look at the "Popular pages" chart for the values of the analytics and forks pages
1. Click on "View forks", then "Back to analytics", then back to forks and back and forth
1. Return to the project analytics page, check that the "Popular pages" chart has higher counts


## Ticket

https://openscience.atlassian.net/browse/EMB-331

# Reviewer Checklist
